ethash/src/cache.rs

```diff
@@ -331,11 +326,8 @@ unsafe fn initialize_memory(memory: *mut Node, num_nodes: usize, ident: &H256) {
 				let mut data: Node = nodes.get_unchecked(data_idx).clone();
 				let rhs: &Node = nodes.get_unchecked(idx);
 
-				unroll! {
-					for w in 0..8 {
-						*data.as_dwords_mut().get_unchecked_mut(w) ^=
-							*rhs.as_dwords().get_unchecked(w);
-					}
+				for (a, b) in data.as_dwords_mut().iter_mut().zip(rhs.as_dwords()) {
```

Comment:
`NodeDoubleWords type` is of length 8? Do this imply bounce checking with the use of `iterator`?

Reply:
https://github.com/paritytech/parity-ethereum/blob/2627288311e952ff3b6b9675dfeda24f75b8249e/ethash/src/shared.rs#L29-L31

Bounds checking is eliminated by LLVM here (see the godbolt link above)